MaterialLib/MPL/Properties/LinearProperty.cpp

```diff
@@ -25,21 +27,38 @@ PropertyDataType LinearProperty::value(
     ParameterLib::SpatialPosition const& /*pos*/,
     double const /*t*/) const
 {
-    return std::get<double>(_value) *
-           (1 +
-            std::get<double>(_independent_variable.slope) *
-                (std::get<double>(variable_array[static_cast<int>(
-                     _independent_variable.type)]) -
-                 std::get<double>(_independent_variable.reference_condition)));
+    auto calculate_linearized_ratio =
+        [&variable_array](double initial_linearized_ratio, auto const& iv) {
+            return initial_linearized_ratio +
+                   std::get<double>(iv.slope) *
+                       (std::get<double>(
+                            variable_array[static_cast<int>(iv.type)]) -
+                        std::get<double>(iv.reference_condition));
+        };
+
+    double linearized_ratio_to_reference_value =
+        std::accumulate(_independent_variables.begin(),
+                        _independent_variables.end(),
+                        1.0,
+                        calculate_linearized_ratio);
+
+    return std::get<double>(_value) * linearized_ratio_to_reference_value;
 }
 
 
 PropertyDataType LinearProperty::dValue(VariableArray const& /*variable_array*/,
                                         Variable const primary_variable) const
 {
-    return _independent_variable.type == primary_variable
+    auto independent_variable =
+        std::find_if(_independent_variables.begin(),
```

Comment:
Maybe you can use an array like
```
using IndependentVariableArray =
    std::array<IndependentVariable, static_cast<int>(Variable::number_of_variables)>;
IndependentVariableArray const _independent_variables; /// With initial value for specified entries
``
Then you can get the value by the entry index of `static_cast<int>( primary_variable).

Reply:
@wenqing If I understood you correctly, instead of a `vector` an `array` should be used with maximum possible variables.
Then we would need to store another list of integeres indicating in which variables the LinearProperty is multi-linear.

Reply:
The additional list of integers wouldn't make the code better readable. And I don't think we would gain much speedup.

Another possibility would be to set the slope of the independent variables in which the property is not linear to zero - but this implies more  computations. I would keep @renchao-lu implementation as it is.

Reply:
By far I keep the implementation as it is.

Reply:
Ok.